### PR TITLE
migrate async-http-client to ce3

### PR DESCRIPTION
**Pull request by yanns**
https://github.com/http4s/http4s/issues/4087

### CODE REVIEW

File: async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala [hunk 1]
```diff
@@ -144,38 +145,43 @@ object AsyncHttpClient {
...
+        dispatcher.unsafeRunSync(
+          onStreamCalled.get
+            .ifM(ifTrue = F.unit, ifFalse = invokeCallbackF[F](cb(Right(response -> dispose))))
+        )
     }
 
   // use fibers to access the ContextShift and ensure that we get off of the AHC thread pool
-  private def invokeCallbackF[F[_]](invoked: => Unit)(implicit F: Concurrent[F]): F[Unit] =
-    F.start(F.delay(invoked)).flatMap(_.join)
+  private def invokeCallbackF[F[_]](invoked: => Unit)(implicit F: Async[F]): F[Unit] =
```
Comment (yanns) [inline async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala:163]: I'm not sure if we still need that.

**Comment by yanns** [comment]
About the test failure:
```
==> X org.http4s.client.asynchttpclient.AsyncHttpClientSpec.AsyncHttpClient Execute GET  30.004s java.util.concurrent.TimeoutException: Futures timed out after [30 seconds]
```

I cannot reproduce it.
`sbt ++2.12.12 test` is running locally.
Maybe an issue due to local configuration, like the number of cores?

**Comment by yanns** [comment]
OK now another test is failing: `org.http4s.client.blaze.Http1ClientStageSuite`

**Comment by m-sp** [comment]
> OK now another test is failing: `org.http4s.client.blaze.Http1ClientStageSuite`:

pretty sure that this is a known flaky one, see https://github.com/http4s/http4s/issues/4151

**Comment by rossabaker** [comment]
Incorrectly autoclosed when cats-effect-3 branch was merged.

**Comment by rossabaker** [comment]
Uh, wait, I think we might have two of these migrations now. See also #4149.

**Comment by rossabaker** [comment]
Yeah, this is a duplicate. I'm sorry for that -- we didn't get a comment or assignment on the original ticket.  The other one is almost done, but I'll merge main into a local copy of this, look at conflict, and make sure there's nothing we missed.

**Comment by rossabaker** [comment]
Yeah, it looks like algebraically equivalent differences, except for some changes to `apply` that let it manage its own `Dispatcher`.

I'm sorry again for the redundancy.  I'll try to keep a closer eye on the ticket management.

**Comment by yanns** [comment]
No problem.

**Pull request closed without merging**